src/freenet/keys/ClientKSK.java

```diff
@@ -43,25 +44,15 @@ public static InsertableClientSSK create(FreenetURI uri) {
 	}
 	
 	public static ClientKSK create(String keyword) {
-		MessageDigest md256 = SHA256.getMessageDigest();
+		byte[] keywordHash = SHA256.digest(keyword.getBytes(StandardCharsets.UTF_8));
+		MersenneTwister mt = new MersenneTwister(keywordHash);
+		DSAPrivateKey privKey = new DSAPrivateKey(Global.DSAgroupBigA, mt);
+		DSAPublicKey pubKey = new DSAPublicKey(Global.DSAgroupBigA, privKey);
+		byte[] pubKeyHash = SHA256.digest(pubKey.asBytes());
 		try {
-			byte[] keywordHash;
-			try {
-				keywordHash = md256.digest(keyword.getBytes("UTF-8"));
-			} catch (UnsupportedEncodingException e) {
-				throw new Error("Impossible: JVM doesn't support UTF-8: " + e, e);
```

Comment:
is this actually impossible with Java 8+?

Reply:
Yes, UTF-8 has been mandatory for all JVMs at least since Java 1.4: https://javaalmanac.io/jdk/1.4/api/java/nio/charset/Charset.html

Reply:
Great! Then getting rid of the catch is long overdue ☺ — nicest way to simplify code is to get rid of no longer needed parts, because the infrastructure improved. :heart: